### DISCUSSION TITLE
Tab扩展按钮 增加一个重新加载当前页面的按钮

### DISCUSSION THREAD

**Discussion by kimdiego2098** [dotnetcore/BootstrapBlazor#3272] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the solution you'd like

业务需求中，可能会需要用户刷新当前页面，才能拿到 其他途径更新的数据

如果需要关闭当前页面，再次打开的话，操作比较繁琐；

**Comment by bb-auto** [comment]
@kimdiego2098 Thank you for contacting us. We will give feedback later.

**Comment by ArgoZhang** [comment]
@kimdiego2098 最开始设计的时候在关闭左边有一个 `Refresh` 按钮的，当时是无法实现这个功能的。目前有什么实现思路吗？

### 需要解决的问题

1. 比如页面有一个下拉框，默认选择的 A，使用者操作更改为 `B` 点击 `Tab` 刷新按钮后是保持 `B` 还是恢复 `A`
2. `Tab` 标签有两种方式，一种是拦截路由把本来是 `Page` 的页面改变成一个标签页，第二种是自定义组件，点击 `Tab` 的刷新按钮有什么机制兼容这两种方式？

**Comment by kimdiego2098** [comment]
@ArgoZhang  事实上我理解的重新加载按钮，并不需要考虑这些，不用保留任何状态，相当于组件/页面释放，重新走一遍组件/页面生命周期

**Reply by ArgoZhang** [reply]
为啥不需要考虑啊？这都是需求啊，可能你不需要，但是别人很可能需要，即使按照你的这个思路走。如何实现呢？有解决方案吗？

**Comment by gitlsl** [comment]
tab页增加个参数 是否持久化state，  yes 的话写入localstorge， 刷新读取 ，  no就不管他拉？
可以扩展为所有页面都支持持久化

**Reply by densen2014** [reply]
你页面自己持久化不更好，这个需求很业务吧

**Reply by densen2014** [reply]
而且tabitem里面本来就是缓存的，你把你有疑虑的代码场景dwmo发出来看看
